core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -665,6 +674,98 @@ public HttpCookie getNonceCookie()
         }
     }
 
+    @DataProvider(name = "groups")
+    public static Object[][] groups()
+    {
+        return new Object[][] {
+                {Optional.empty()},
+                {Optional.of(ImmutableSet.of())},
+                {Optional.of(ImmutableSet.of("admin", "public"))}
+        };
+    }
+
+    @Test(dataProvider = "groups")
+    public void testOAuth2Groups(Optional<Set<String>> groups)
+            throws Exception
+    {
+        try (TokenServer tokenServer = new TokenServer(Optional.empty());
+                TestingTrinoServer server = TestingTrinoServer.builder()
+                        .setProperties(ImmutableMap.<String, String>builder()
+                                .putAll(SECURE_PROPERTIES)
+                                .put("http-server.authentication.type", "oauth2")
+                                .put("web-ui.enabled", "true")
+                                .put("http-server.authentication.oauth2.issuer", tokenServer.getIssuer())
+                                .put("http-server.authentication.oauth2.jwks-url", tokenServer.getJwksUrl())
+                                .put("http-server.authentication.oauth2.state-key", "test-state-key")
+                                .put("http-server.authentication.oauth2.auth-url", tokenServer.getIssuer())
+                                .put("http-server.authentication.oauth2.token-url", tokenServer.getIssuer())
+                                .put("http-server.authentication.oauth2.client-id", tokenServer.getClientId())
+                                .put("http-server.authentication.oauth2.client-secret", tokenServer.getClientSecret())
+                                .put("http-server.authentication.oauth2.groups-field", GROUPS_CLAIM)
+                                .build())
+                        .setAdditionalModule(binder -> {
+                            jaxrsBinder(binder).bind(TestResource.class);
+                            newOptionalBinder(binder, OAuth2Client.class)
+                                    .setBinding()
+                                    .toInstance(tokenServer.getOAuth2Client());
+                        })
+                        .build()) {
+            server.getInstance(Key.get(AccessControlManager.class)).addSystemAccessControl(TestSystemAccessControl.NO_IMPERSONATION);
```

Comment:
Why no impersonation?

Reply:
It's not important actually whether it allows impersonation or not but I think it's better to deny by default.
The important part is:
```java
@Override
public void checkCanReadSystemInformation(SystemSecurityContext context)
{
    if (!context.getIdentity().getUser().equals(MANAGEMENT_USER)) {
        denyReadSystemInformationAccess();
    }
}
```
as it's being tested by `assertAuthenticationAutomatic`